tests/components/tplink/test_init.py

```diff
@@ -700,6 +715,10 @@ async def test_credentials_hash_auth_error(
 
     with (
         patch("homeassistant.components.tplink.PLATFORMS", []),
+        patch(
+            "homeassistant.components.tplink.async_create_clientsession",
+            return_value="Foo",
```

Comment:
as above

Reply:
This isn't great but we are patching the API being exposed to the integration so its less concerning.  We should clean this up in the future so we don't make this PR any larger as it likely will need another lib bump to make it patchable in the lib, and I'm already concerned about the size of this PR for a patch.  I think we need to do it for a patch given the impact of the problem though.